app/controllers/core_data_connector/names_controller.rb

```diff
@@ -0,0 +1,43 @@
+module CoreDataConnector
+  class NamesController < ApplicationController
+    # Search attributes
+    search_attributes :name
+
+    def base_query
+      query = super
+
+      return query.none unless params[:project_id]
+
+      nameable_models = [
+        :instance,
+        :item,
+        :work
+      ]
+
+      queries = nameable_models.map do |nm|
+        model_name = "CoreDataConnector::#{nm.to_s.classify}".constantize
+
+        CoreDataConnector::SourceTitle
```

Comment:
Since we're in the `CoreDataConnector` module, no need to prefix the class here.